platform/android/MapboxGLAndroidSDK/src/main/java/org/maplibre/android/annotations/AnnotationContainer.kt

```diff
@@ -82,13 +94,21 @@ class KAnnotationContainer(
     private fun groupAnnotations(): Map<Key, List<KAnnotation<*>>> =
         annotationList.groupBy { it.key() }
 
-    private data class Key(private val type: Class<in KAnnotation<*>>) // TODO will be expanded in the future
+    private data class Key(val type: KClass<out KAnnotation<*>>) // TODO will be expanded in the future
 
-    private fun KAnnotation<*>.key() = Key(this.javaClass)
+    private fun KAnnotation<*>.key() = Key(this::class)
 
     private fun MutableMap<Key, AnnotationManager<*, *>>.getOrCreate(key: Key): AnnotationManager<*, *>? =
         get(key) ?: style?.let {
-            SymbolManager(mapView, mapLibreMap, it)
+            when (key.type) {
+                Symbol::class -> SymbolManager(mapView, mapLibreMap, it)
+                Circle::class -> CircleManager(mapView, mapLibreMap, it)
+                Line::class -> LineManager(mapView, mapLibreMap, it)
+                Fill::class -> FillManager(mapView, mapLibreMap, it)
+                else -> throw IllegalArgumentException(
+                    "Impossible key! This should never occur because KAnnotation is a sealed class."
```

Comment:
Is this needed when it is a sealed class?

Reply:
@louwers From what I've tried, yes. `key.type` is not of type `KAnnotation` (unlike the other `when` statements in this file), and it seems like the compiler can't derive from the fact that `key.type` is of type `KClass<KAnnotation<*>>` that there's only a limited number of (equivalence groups of) instances that could be given as inputs.